radiography/src/main/java/radiography/StateRenderer.kt

```diff
@@ -0,0 +1,36 @@
+package radiography
+
+import radiography.StateRenderer.Companion.from
+
+/**
+ * Renders extra attributes for specifics types in the output of [Radiography.scan].
+ * Call [from] to create instances with reified types:
+ * ```
+ *  val myRenderer: StateRenderer<MyView> = from { myView, append ->
+ *    append(myView.customAttributeValue)
+ *  }
+ * ```
+ */
+abstract class StateRenderer<T>(
```

Comment:
Maybe `ViewStateRenderer`?

Reply:
I've been thinking we could almost move this into the more generic TreeStringRenderer? Hence it works with anything that has state.

Also annoying that one of those ViewStateRenderer would be the View class ViewStateRenderer